lib/virtio/virtqueue.c

```diff
@@ -180,7 +180,7 @@ void *virtqueue_get_buffer(struct virtqueue *vq, uint32_t *len, uint16_t *idx)
 
 	atomic_thread_fence(memory_order_seq_cst);
 
-	desc_idx = (uint16_t)uep->id;
+	desc_idx = uep->id;
```

Comment:
uint16_t casting is required as the uep->id is uint32_t, but the desc_idx is uint16_t.

Reply:
Fix